shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/presentation/detail/PlantDetailComponent.kt

```diff
@@ -10,24 +11,32 @@ import kotlinx.coroutines.flow.update
 class PlantDetailComponent(
     plant: Plant,
     componentContext: ComponentContext,
-    private val onNavigateToBack: ()-> Unit
-): ComponentContext by componentContext {
+    private val plantsRepository: PlantsRepository,
+    private val onNavigateToBack: () -> Unit
+) : ComponentContext by componentContext {
     private val _state = MutableStateFlow(PlantDetailState(plant))
```

Comment:
If you accepted my previous suggestion then you can combine favourite StateFlow from repository here with the uiState